tracer/src/Datadog.Trace/Configuration/ExtraServicesProvider.cs

```diff
@@ -0,0 +1,47 @@
+﻿// <copyright file="ExtraServicesProvider.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Configuration;
+
+internal class ExtraServicesProvider : IExtraServicesProvider
+{
+    private const int MaxExtraServices = 64;
+    private const string FakeValue = null;
+
+    // no concurrent hash set, so use a dictionary with empty values
+    private readonly ConcurrentDictionary<string, string> _extraServices = new();
```

Comment:
IIRC, service names are all normalized to be lower case in the backend, so should we use a `StringComparer.OrdinalIgnoreCase` here 🤔

Reply:
Fixed see 14f828b6ca89a2f776967e35a647b6493ec40c97